src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form.ts

```diff
@@ -316,40 +264,91 @@ export function useForm<T extends FormData = FormData, I extends FormData = T>(
         return prev;
       });
     },
-    [getFormData$, fieldsToArray]
+    [getFormData$, updateFormData$, fieldsToArray]
+  );
+
+  const getFieldDefaultValue: FormHook<T, I>['__getFieldDefaultValue'] = useCallback(
+    (fieldName) => get(defaultValueDeserialized.current, fieldName),
+    []
+  );
+
+  const readFieldConfigFromSchema: FormHook<T, I>['__readFieldConfigFromSchema'] = useCallback(
+    (fieldName) => {
+      const config = (get(schema ?? {}, fieldName) as FieldConfig) || {};
+
+      return config;
+    },
+    [schema]
   );
 
-  const setFieldValue: FormHook<T>['setFieldValue'] = useCallback((fieldName, value) => {
+  // ----------------------------------
+  // -- Public API
+  // ----------------------------------
+  const getFormData: FormHook<T, I>['getFormData'] = useCallback(() => {
+    const fieldsToOutput = getFieldsForOutput(fieldsRefs.current, {
+      stripEmptyFields: formOptions.stripEmptyFields,
+    });
+    const fieldsValue = mapFormFields(fieldsToOutput, (field) => field.__serializeValue());
+    return serializer
+      ? serializer(unflattenObject<I>(fieldsValue))
+      : unflattenObject<T>(fieldsValue);
```

Comment:
So the form lib, internally still stores paths as `my.path[0].to.something`? We just don't expose this to consumer code?

Reply:
Yes. I thought in changing that but when I started to write the logic to remove a field from an array (e.g. second position) from inside an object (at any depth) I thought... let's keep it flat until the last minute! 😄 It makes the internal logic much simpler.